.github/workflows/linux_ci.yml

```diff
@@ -185,11 +251,22 @@ jobs:
     - name: Install tesseract
       run: sudo apt-get install tesseract-ocr libtesseract-dev poppler-utils
 
+    - name: Install Dependencies (on cache miss only)
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: |
+        pip install --upgrade pip
+        pip install .[test]
+        pip install rest_api/
+        pip install ui/
+        pip install torch-scatter -f https://data.pyg.org/whl/torch-1.10.0+cpu.html
+        echo "=== pip freeze ==="
+        pip freeze
+
     - name: Reinstall Haystack
       run: |
         pip install .[test]
         pip install rest_api/
-        pip install eager ui/
+        pip install ui/
 
     - name: Run tests
-      run: pytest -s ${{ matrix.test-path }}
+      run: pytest -s --random-order ${{ matrix.test-path }}
```

Comment:
I wouldn't introduce additional randomness here. This could be super confusing to analyze and make some tests fail due to temporary memory pressure. Did we get any value from that?

Btw, in order to better analyze tests and test output, we could set `log_cli = True` in setup.cfg's pytest section.